src/app/modules/main/chat_section/chat_content/users/module.nim

```diff
@@ -47,27 +50,41 @@ method isLoaded*(self: Module): bool =
 
 method viewDidLoad*(self: Module) =
   # add me as the first user to the list
+  let (admin, joined) = self.controller.getChatMemberInfo(singletonInstance.userProfile.getPubKey())
   let loggedInUserDisplayName = singletonInstance.userProfile.getName() & "(You)"
-  self.view.model().addItem(initItem(singletonInstance.userProfile.getPubKey(), loggedInUserDisplayName, 
-  OnlineStatus.Online, singletonInstance.userProfile.getIcon(), singletonInstance.userProfile.getIsIdenticon()))
-
-  # add other memebers
-  let usersKeys = self.controller.getMembersPublicKeys()
-  for k in usersKeys:
-    if (k == singletonInstance.userProfile.getPubKey()):
+  self.view.model().addItem(initItem(
+    singletonInstance.userProfile.getPubKey(), 
+    loggedInUserDisplayName, 
+    OnlineStatus.Online, 
+    singletonInstance.userProfile.getIcon(), 
+    singletonInstance.userProfile.getIsIdenticon(),
+    admin,
+    joined,
+  ))
+
+    # add other memebers
+  let publicKeys = self.controller.getMembersPublicKeys()
+  for publicKey in publicKeys:
+    if (publicKey == singletonInstance.userProfile.getPubKey()):
       continue
-    let (name, image, isIdenticon) = self.controller.getContactNameAndImage(k)
-    let statusUpdateDto = self.controller.getStatusForContact(k)
+
+    let (admin, joined) = self.controller.getChatMemberInfo(publicKey)
+    let (name, image, isIdenticon) = self.controller.getContactNameAndImage(publicKey)
+    let statusUpdateDto = self.controller.getStatusForContact(publicKey)
     let status = statusUpdateDto.statusType.int.OnlineStatus
-    self.view.model().addItem(initItem(k, name, status, image, isidenticon))
+    self.view.model().addItem(initItem(publicKey, name, status, image, isidenticon, admin, joined))
 
   self.moduleLoaded = true
   self.delegate.usersDidLoad()
 
 method getModuleAsVariant*(self: Module): QVariant =
   return self.viewVariant
 
-method newMessagesLoaded*(self: Module, messages: seq[MessageDto]) = 
+method newMessagesLoaded*(self: Module, messages: seq[MessageDto]) =
+  let chat = self.controller.getChat()
+  if chat.isPrivateGroupChat():
```

Comment:
We could even make that check to return if it's **not** a public chat? Since it's only public chats that add users on new messages

Reply:
good point, updated